src/mlpack/bindings/R/print_type_doc_impl.hpp

```diff
@@ -0,0 +1,165 @@
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& data,
+    const typename std::enable_if<util::IsStdVector<T>::value>::type*)
+{
+  if (std::is_same<T, std::vector<int>>::value)
+  {
+    return "A vector of integers; i.e., c(0, 1, 2).";
+  }
+  else if (std::is_same<T, std::vector<std::string>>::value)
+  {
+    return "A vector of strings; i.e., c(\"hello\", \"goodbye\").";
+  }
+  else
+  {
+    throw std::invalid_argument("Unknown vector type '" + data.cppType + "'.");
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a matrix option.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& data,
+    const typename std::enable_if<arma::is_arma_type<T>::value>::type*)
+{
+  if (std::is_same<typename T::elem_type, double>::value)
+  {
+    if (T::is_col || T::is_row)
+    {
+      return "A 1-d matrix-like containing 'numeric' data (could be an "
+          "'matrix' or 'data.frame' with one dimension of size 1).";
+    }
+    else
+    {
+      return "A 2-d matrix-like containing 'numeric' data (could be an "
+          "'matrix' or a 'data.frame' or anything convertible to an "
+          "2-d 'matrix'.";
+    }
+  }
+  else if (std::is_same<typename T::elem_type, size_t>::value)
+  {
+    if (T::is_col || T::is_row)
+    {
+      return "A 1-d matrix-like containing 'integer' data (could be an "
+          "'matrix' or 'data.frame' with one dimension of size 1).";
+    }
+    else
+    {
+      return "A 2-d matrix-like containing 'integer' data (could be an "
+          "'matrix' or a 'data.frame' or anything convertible to an "
+          "2-d 'matrix'.";
+    }
+  }
+  else
+  {
+    throw std::invalid_argument("Unknown matrix type '" + data.cppType + "'.");
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a matrix tuple option.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& /* data */,
+    const typename std::enable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>::value>::type*)
+{
+  return "A 2-d array containing 'numeric' data.  Like the regular 2-d matrices"
+      ", this can be a 'matrix', or a data.frame. However, this type can also "
+      "accept a data.frame that has columns of type 'character', 'logical' or "
+      "'factor'.  These values will be converted to numeric indices before "
+      "being passed to mlpack, and then inside mlpack they will be properly "
+      "treated as categorical variables, so there is no need to do one-hot "
+      "encoding for this matrix type.";
+}
+
+/**
+ * Return a string representing the command-line type of a model.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& /* data */,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::enable_if<data::HasSerialize<T>>::type*)
+{
+  return "An mlpack model pointer.  '<Model>' refers to the type of model that "
```

Comment:
This doesn't display quite right:

http://mlpack2.ratml.org/doc/mlpack-git/r_documentation.html#data-formats

I think that if you use backticks for types here, everything should work.  Take a look at `mlpack::bindings::julia::PrintTypeDoc()` to see what I mean.

Reply:
I have updated this. Thank you for noticing this.

Reply:
Looks better now.  Thanks! :+1: http://mlpack2.ratml.org/doc/mlpack-git/r_documentation.html#data-formats